### DISCUSSION TITLE
Copilot cannot index vault on Gemini-1.5-flash

### DISCUSSION THREAD

**Discussion by TaiRaven** [logancyang/obsidian-copilot#721] [Q&A] [answered]
Hi there, 

Due to my works policies on ChatGPT, I am forced to use Gemini 1.5 Flash as my LLM of choice. 

I have connected this, and can chat and send the current note, but clicking the Index vault option causes the indexing to stop at 0/1000 notes. I have run the force re-index vault command, but that has not worked. 

![image](https://github.com/user-attachments/assets/739ce2d9-67c3-41ab-8f0e-f2d1831da150)

This srror pops up, "Error indexing vault: API Key is not provided for the embedding model.

Assuming then that Gemini cannot index the vault?

**Comment by logancyang** [comment, marked as answer]
Please check your QA setting, you need a working embedding model and your current embedding model is probably set to an OpenAI one. Just use Google's embedding model `text-embedding-004`, it should work.
